src/Collection/Fulfilment/OrderNotesCollection.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Sdk\src\Collection\Fulfilment;
+
+use MyParcelNL\Sdk\src\Concerns\HasApiKey;
+use MyParcelNL\Sdk\src\Concerns\HasUserAgent;
+use MyParcelNL\Sdk\src\Model\Fulfilment\OrderNote;
+use MyParcelNL\Sdk\src\Model\MyParcelRequest;
+use MyParcelNL\Sdk\src\Model\RequestBody;
+use MyParcelNL\Sdk\src\Support\Arr;
+use MyParcelNL\Sdk\src\Support\Collection;
+
+class OrderNotesCollection extends Collection
+{
+    use HasUserAgent;
+    use HasApiKey;
+
+    /**
+     * @return array Indexed array of order uuids present in this collection.
+     */
+    private function orderUuidsInCollection(): array
+    {
+        return array_unique($this->reduce(
+            function (array $uuids, OrderNote $orderNote) {
+                $uuids[] = $orderNote->getOrderUuid();
+
+                return $uuids;
+            },
+            []
+        ));
+    }
+
+    /**
+     * @return self Collection of notes that were saved.
+     *
+     * @throws \MyParcelNL\Sdk\src\Exception\AccountNotActiveException
+     * @throws \MyParcelNL\Sdk\src\Exception\ApiException
+     * @throws \MyParcelNL\Sdk\src\Exception\MissingFieldException
+     * @throws \Exception
+     */
+    public function save(): self
+    {
+        $notes = [];
```

Comment:
als je `$this->orderUuidsInCollection()` hebt aangepast dat het een collection teruggeeft kan je `Collection::reduce` gebruiken en dit wat korter maken, bijvoorbeeld:

Reply:
Ik heb het inderdaad anders geschreven. Zelf vind ik het verwarrend om de collection te gebruiken voor een lijstje uuid’s (ik verwacht dan toch ordernotes erin), of ik heb het verkeerd begrepen.